numpy/core/src/multiarray/iterators.c

```diff
@@ -666,9 +666,6 @@ iter_subscript(PyArrayIterObject *self, PyObject *ind)
         Py_DECREF(obj);
         return (PyObject *)ret;
     }
-    else {
```

Comment:
This is very nicely found!

I would prefer, though, to correct it a bit differently: change the `if` statement on line 643 to:
```
if (!PyArray_Check(obj)) {
    goto fail;
}
```
and then un-indent what is inside the current `if` clause. That way, the "good" part clearly ends with a `return` before the `fail` branch. What do you think?

Reply:
Of course, you're right, that looks much cleaner, thank you. But then I think I'll slightly refactor that block further, the disconnect between the return on line 666 and its direct predecessor on line 647 stuck out to me when I first looked at it.